integrationTests/CreateDonationWithPersistErrorTest.php

```diff
@@ -0,0 +1,141 @@
+ * To solve this issue:
+ * Either explicitly call EntityManager#persist() on this unknown entity or configure cascade persist this association
+ * in the mapping for example @ManyToOne(..,cascade={"persist"}).
+ */
+class CreateDonationWithPersistErrorTest extends IntegrationTest
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $fundsReturnedFromSfAPI = [
+            [
+                'id' => 'fnsfid',
+                'type' => 'pledge',
+                'totalAmount' => '1000',
+                'isShared' => false,
+                'currencyCode' => 'GBP',
+                'amountForCampaign' => '1000',
+            ],
+        ];
+
+        $campaignReturnedFromSfAPI = [
+            'id' => '1',
+            'currencyCode' => 'GBP',
+            'endDate' => '2023-11-11T17:21:50+00:00',
+            'startDate' => '2023-09-11T17:21:50+00:00',
+            'feePercentage' => 1.0,
+            'isMatched' => true,
+            'title' => 'Save the PHP Developers',
+            'charity' => [
+                'id' => '',
+                'name' => 'Some Charity',
+                'stripeAccountId' => 'stripe-acc-id',
+                'giftAidOnboardingStatus' => 'Onboarded',
+                'hmrcReferenceNumber' => '',
+                'regulatorRegion' => '',
+                'regulatorNumber' => '',
+            ],
+        ];
+
+        $donationClientProphecy = $this->prophesize(\MatchBot\Client\Donation::class);
+        $donationClientProphecy->create(Argument::type(Donation::class))->willReturn($this->randomString());
+
+        $fundClientProphecy = $this->prophesize(\MatchBot\Client\Fund::class);
+
+        $fundClientProphecy->getForCampaign(Argument::type('string'))->willReturn($fundsReturnedFromSfAPI);
+
+        $campaignClientProphecy = $this->prophesize(\MatchBot\Client\Campaign::class);
+        $campaignClientProphecy->getById(Argument::type('string'))->willReturn(
+            $campaignReturnedFromSfAPI
+        );
+
+        /**
+         * the error is only thrown if this line is here. Query could we somehow have two different entity managers
+         * at once in prod and could that be causing the problem? Our RetrySafeEntityManager and the standard doctrine
+         * one as below?
+         *
+         * @psalm-suppress DeprecatedMethod
+         * @psalm-suppress MixedArrayAccess
+         * @psalm-suppress MixedArgument
+         * We're using EntityManager::create already inside RetrySafeEntityManager - while we have it there we may as
+         * well have it in this test. Also probably not worthwhile fixing the Mixed issues here before they're
+         * fixed in the similar prod code.
+         */
+        $this->setInContainer(EntityManagerInterface::class, EntityManager::create(
+            // for this test we don't need the RetrySafeEntityManager - using the standard EM makes things simpler.
```

Comment:
actually this is misleading - in this test the crash only happens because have two different entity manages due to this line. In real scenarios we probably have a replaced underlying em because the original one closed. I been testing and trying to fix that scenario on my local.